plugins/env/app/models/environment_variable_group.rb

```diff
@@ -0,0 +1,18 @@
+class EnvironmentVariableGroup < ActiveRecord::Base
+  default_scope -> { order(:name) }
+
+  has_many :environment_variables
+  has_many :stage_environment_variable_groups
+  has_many :stages, through: :stage_environment_variable_groups
```

Comment:
I prefer concrete models over habtm since most of the time some logic ends
up there anyway + it's 1 less abstraction / way of handling things and
allows me to do some efficient queries like stage_environment_variable_
groups.pluck(:stage_id)

On Thu, May 28, 2015 at 8:03 AM, Shane Hender <notifications@github.com>
wrote:

> In plugins/env/app/models/environment_variable_group.rb
> <https://github.com/zendesk/samson/pull/397#discussion_r31242464>:
>
> > @@ -0,0 +1,18 @@
> > +class EnvironmentVariableGroup < ActiveRecord::Base
> > +  default_scope -> { order(:name) }
> > +
> > +  has_many :environment_variables
> > +  has_many :stage_environment_variable_groups
> > +  has_many :stages, through: :stage_environment_variable_groups
>
> You don't have to specify the through table if you use the HABTM
> relationship. See DeployGroups to Stages
>
>   has_and_belongs_to_many :environment_variable_groups
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/397/files#r31242464>.
>